renderdoc/driver/shaders/dxbc/dxbc_bytecode.cpp

```diff
@@ -380,16 +380,36 @@ DXBC::Reflection *Program::GuessReflection()
         cb.descriptor.flags = 0;
         cb.descriptor.numVars = numVecs;
 
-        cb.variables.reserve(numVecs);
+        bool isArray = desc.bindCount > 1;
+        if(isArray)
+        {
+          // If the constant buffer is an array, then we need to add an entry for the struct
+          // itself. This mimics what is loaded for a constant buffer array when reflection
+          // information is not stripped.
+          DXBC::CBufferVariable var;
+          var.name = cb.name;
+          var.descriptor.name = cb.name;
+          var.descriptor.offset = 0;
+          var.descriptor.flags = 2;
```

Comment:
Is this 2 by inspection from reflection data, or does it have a known meaning?

Reply:
Both. The reflected data contains a 2 for this struct variable. This flag corresponds to `D3D_SHADER_VARIABLE_FLAGS` where 2 indicates that the variable is used by the shader.